addons/turbowarp-player/userscript.js

```diff
@@ -27,12 +27,19 @@ export default async function ({ addon, console, msg }) {
   }
 
   button.onclick = async () => {
+    const projectId = window.location.pathname.split("/")[2];
     let search = "";
-    if (
-      addon.tab.redux.state?.preview?.projectInfo?.public === false &&
-      addon.tab.redux.state.preview.projectInfo.project_token
-    ) {
-      search = `#?token=${addon.tab.redux.state.preview.projectInfo.project_token}`;
+    if (addon.tab.redux.state?.preview?.projectInfo?.public === false) {
+      let projectToken = (
+        await (
+          await fetch(`https://api.scratch.mit.edu/projects/${projectId}?nocache=${Date.now()}`, {
```

Comment:
Will change the URL parameter. See #6717

Reply:
Also, if this request takes too long, the browser will think it's blocking a pop-up when trying to call `window.open()` below.

Reply:
My PR: #6728